### ISSUE TITLE
Unable to save a document field with 'undefined' value

### ISSUE THREAD

**Issue by arggh** [jagi/meteor-astronomy#652]
This doesn't work:

```
const doc = Doc.findOne();
doc.title = undefined;
doc.save();
```
If you replace `undefined` with `null` it works as expected. I have tried `forceUpdate` and defining the specific fields to update with `fields: ['title']` with no luck.

Reproduction: https://github.com/arggh/astronomy-undefined-issue

**Comment by arggh** [comment]
The issue is in `/lib/modules/storage/utils/apply_modifier.js`.

```js
    const rawDoc = doc.raw();
    // Use Minimongo's the "_modify" method to apply modifier.
    LocalCollection._modify(rawDoc, modifier, options);
    // Set all values back again on a document.
    doc.set(rawDoc, {
      defaults: config.defaults,
      clone: false,
      cast: false
    });
```
#### What this does:

1. it applies the modifier to the `rawDoc`, which correctly doesn't have the removed field after calling `_modify`.

2. Then, the modified raw doc gets _applied_ or _set_ on the original doc, which still has the removed field intact...

3. ...which means, it never gets removed / unset, except momentarily from the temporary raw doc.

**Comment by lukejagodzinski** [comment]
Hey, thanks for posting this issue. Indeed, there seems to be an error probably because of the change in how Meteor's `LocalCollection._modify` function works. I'm already working on the fix. It should be ready today.

You were almost right in how you've implemented your PR but there is already an utility function for what you've tried to do :)

**Comment by lukejagodzinski** [comment]
Fixed in `jagi:astronomy@2.5.3`
